core/input/input.h

```diff
@@ -182,17 +182,17 @@ class Input : public Object {
 	struct JoyBinding {
 		JoyType inputType;
 		union {
-			int button;
+			JoyButton button;
 
 			struct {
-				int axis;
+				JoyAxis axis;
 				JoyAxisRange range;
 				bool invert;
 			} axis;
 
 			struct {
-				int hat;
-				HatMask hat_mask;
+				HatDir hat;
+				int hat_mask;
```

Comment:
Is it by mistake: `HatMask -> int`?

Reply:
This was discussed in the PR meeting today. Masks can have multiple bits enabled, so the type should be int.

Reply:
I think the existing type should be kept here though. Code that assigns to it even static casts to `HatMask`.